store/postgres/src/relational_queries.rs

```diff
@@ -1291,24 +1293,50 @@ impl<'a> QueryFragment<Pg> for InsertQuery<'a> {
         out.push_sql(self.table.qualified_name.as_str());
 
         out.push_sql("(");
-        for column in self.table.columns.iter() {
-            if self.entity.contains_key(&column.field) {
-                out.push_identifier(column.name.as_str())?;
-                out.push_sql(", ");
+
+        // Build the column name list using the subset of all keys among our entities.
+        let unique_column_names: Vec<&Column> = {
+            let mut btreemap = BTreeMap::new();
+            for (_key, entity) in self.entities.iter() {
+                for column in &self.table.columns {
+                    if entity.get(&column.field).is_some() {
+                        btreemap.entry(column.name.as_str()).or_insert(column);
+                    }
+                }
```

Comment:
The computation of `unique_columns` should go into `new` since `walk_ast` can be called multiple times by Diesel, and we'd redundantly repeat the calculation.